### DISCUSSION TITLE
trying to connect to jdtls (JAVA) via socat on remote system to avoid tramp lock

### DISCUSSION THREAD

**Discussion by thkoch2001** [joaotavora/eglot#1170] [Q&A] [answered]
Right now, eglot over tramp seems to not work ([tramp bug](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=61350)).

Unfortunately, jdtls does also not support to act as a tcp server ([jdtls issue](https://github.com/eclipse/eclipse.jdt.ls/issues/431)). It expects the editor instead to act as the server.

So I thought I could forward the jdtls stdin and stdout with socat:

    socat -d -d EXEC:jdtls TCP-LISTEN:2223

With `C-u M-x eglot` I can provide `localhost:2223`. Unfortunatly eglot claims 

> Process EGLOT (yacy_search_server/(java-mode java-ts-mode)) not running

I tested with telnet that port 2223 is indeed forwarded from my VM to the host running emacs.

**Comment by joaotavora** [comment, marked as answer]
This is worked around in latest Eglot.  The Tramp fix is being discussed in Emacs bug#61350. https://debbugs.gnu.org/cgi/bugreport.cgi?bug=61350
